maven-release-manager/src/main/java/org/apache/maven/shared/release/transform/jdom2/JDomProperties.java

```diff
@@ -51,10 +51,13 @@ public JDomProperties(Element properties) {
     }
 
     @Override
-    public synchronized Object setProperty(String key, String value) {
-        Element property = properties.getChild(key, properties.getNamespace());
+    public synchronized Object put(Object key, Object value) {
```

Comment:
I would rather tend to rewrite this to just implement `Map<String,String>`. By extending from `Properties` without ever populating the underlying data structure from `Hashtable` this is very error prone. Lots of methods like `computeIfPresent()` will just not work correctly.

Reply:
As org.apache.maven.model.ModelBase unfortunately exposes `java.util.Properties` this is not easily possible. Need to think a bit more how to do this without major refactoring...